CadRevealFbxProvider.Tests/BatchUtils/ScaffoldOptimizerTests.cs

```diff
@@ -98,34 +101,53 @@ List<uint> truthIndices
         };
     }
 
-    private void CheckGeometries(APrimitive[] primitives, List<Vector3> truthVertices, List<uint> truthIndices)
+    private static void CheckGeometries(
+        APrimitive[] primitives,
+        List<Vector3> truthVertices,
+        List<uint> truthIndices,
+        List<BoundingBox> originalBoundingBoxes
+    )
     {
-        foreach (var primitive in primitives)
+        Assert.That(originalBoundingBoxes.ToArray(), Has.Length.EqualTo(primitives.Length));
```

Comment:
nit. toArray() should be unneccessary in this assertion. You can do `Has.Exactly(primitives.Length).Items`